### PR TITLE
Show errors in recipe resolution in the dev-shell

### PR DESCRIPTION
**Pull request by Cypher1**

### CODE REVIEW

File: shells/dev-shell/index.js [hunk 1]
```diff
@@ -73,9 +73,12 @@ async function wrappedExecute() {
     arcPanel.attachArc(arc);
 
     const resolver = new RecipeResolver(arc);
-    const resolvedRecipe = await resolver.resolve(recipe);
+    const options = {errors: new Map()};
+    const resolvedRecipe = await resolver.resolve(recipe, options);
```
Comment (mykmartin) [inline shells/dev-shell/index.js]: `resolve()` doesn't take an options object.

File: shells/dev-shell/index.js [hunk 2]
```diff
@@ -73,9 +73,12 @@ async function wrappedExecute() {
...
     if (!resolvedRecipe) {
-      arcPanel.showError('Error in RecipeResolver');
+      arcPanel.showError(`Could not normalize a context recipe:\n${
+        [...options.errors.entries()].join('\n')
+      }.\n${recipe.toString()}`);
```
Comment (mykmartin) [inline shells/dev-shell/index.js]: The error panel has a header and separate display for details, so this should just be `arcPanel.showError('Error resolving recipe', recipe.toString());`

File: shells/dev-shell/output-pane.js [hunk 3]
```diff
@@ -223,7 +223,7 @@ const errorTemplate = `
     }
   </style>
   <div class="container">
-    <div id="header"></div>
+    <pre id="header"></pre>
```
Comment (mykmartin) [inline shells/dev-shell/output-pane.js]: No need to change this; the `<pre id="message">` element should show the recipe.

File: shells/dev-shell/index.js (hunk 1 above)
Comment (Cypher1) [reply shells/dev-shell/index.js]: I believe the follow is the current implementation (which does take a options value.

src/runtime/recipe/recipe-resolver.ts:179
```
  // Attempts to run basic resolution on the given recipe. Returns a new
  // instance of the recipe normalized and resolved if possible. Returns null if
  // normalization or attempting to resolve slot connection fails.
  async resolve(recipe: Recipe, options?: IsValidOptions ) {
```

Comment (Cypher1) [reply shells/dev-shell/index.js]: Ah, managed to split my commits incorrectly. Uploading the rest of the change.

File: shells/dev-shell/output-pane.js (hunk 3 above)
Comment (Cypher1) [reply shells/dev-shell/output-pane.js]: Neat, changed.

File: shells/dev-shell/index.js (hunk 2 above)
Comment (Cypher1) [reply shells/dev-shell/index.js]: Thanks (done).

**Pull request merged**